### PR TITLE
Issue #7601: Add examples for JavadocType

### LINKED ISSUE #7601: Update doc for JavadocType
https://checkstyle.org/config_javadoc.html#JavadocType

Add example for each existing config to show what code is ok and what code cause violation.
Good example is https://checkstyle.org/config_naming.html#AbstractClassName
Look at referenced commits of #5832, to see what and how should be changed in code.

Please play with module/check by means of https://checkstyle.org/cmdline.html to understand how it works.

------------

PLEASE READ https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

### PR DESCRIPTION
**Pull request by HCY123902**
Resolves #7601: Add examples for JavadocType

### CODE REVIEW

**Comment by HCY123902** [comment]
Github, generate web site

**Comment by HCY123902** [comment]
Github, generate web site

**Comment by nrmancuso** [comment]
@HCY123902 please generate the site locally to avoid wasting CI resources, see https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-website-only

**Comment by HCY123902** [comment]
> @HCY123902 please generate the site locally to avoid wasting CI resources, see https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-website-only

yeah sure. Thanks for the suggestion. This is my first pull request and I may still need to get more familiar with the contribution procedure. The CI problems are caused by `slices_should_be_free_of_cycles_suppressions`, which I think is caused by the additional Javadoc added

**Comment by nrmancuso** [comment]
@HCY123902 you must run mvn clean verify locally to generate changes in those files, and you should always run this command and make sure it is passing before pushing to OR

**Comment by HCY123902** [comment]
Example results

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType"/>
    </module>
</module>
```

```
/**
 * @unknownTag value // violation
 */
public class PublicClass {}

/**
 * @unknownTag value // violation
 */
private class PrivateClass {}
```
Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_one_checks.xml -o example_one_result.out ./src/ExampleOne.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleOne.java:2:4: 未知标签 'unknownTag' 。 [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleOne.java:7:4: 未知标签 'unknownTag' 。 [JavadocType]
检查完成。
```
---
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="scope" value="public"/>
        </module>
    </module>
</module>
```

```
/**
 * @unknownTag value // violation
 */
public class PublicClass {}

/**
 * @unknownTag value // OK
 */
private class PrivateClass {} // OK
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_two_checks.xml -o example_two_result.out ./src/ExampleTwo.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleTwo.java:2:4: 未知标签 'unknownTag' 。 [JavadocType]
检查完成。
```
---
```
/**
 * @author a // OK
 */
public class ClassA { // OK
    /**
     */
    private class ClassB {} // OK, since author format check is ignored for inner class
}

/**
 */
public class ClassC {} // violation, author tag is missing

public class ClassD {} // OK, since it does not have JavaDoc
```

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="authorFormat" value="\S"/>
        </module>
    </module>
</module>
```

```
/**
 * @author a // OK
 */
public class ClassA { // OK
    /**
     */
    private class ClassB {} // OK, since author format check is ignored for inner class
}

/**
 */
public class ClassC {} // violation, author tag is missing

public class ClassD {} // OK, since it does not have JavaDoc
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_three_checks.xml -o example_three_result.out ./src/ExampleThree.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleThree.java:12:1: 缺少 @author 标签。 [JavadocType]
检查完成。
```
---
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="scope" value="private"/>
            <property name="excludeScope" value="package"/>
        </module>
    </module>
</module>
```

```
/**
 * @unknownTag value // OK
 */
public class PublicClass {} // OK

/**
 * @unknownTag value // OK
 */
protected class ProtectedClass {} // OK

/**
 * @unknownTag value // OK
 */
class PackagePrivateClass {} // OK

/**
 * @unknownTag value // violation
 */
private class PrivateClass {}
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_five_checks.xml -o example_five_result.out ./src/ExampleFive.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleFive.java:17:4: 未知标签 'unknownTag' 。 [JavadocType]
检查完成。
```
---
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="allowedAnnotations" value="SpringBootApplication,Configuration"/>
        </module>
    </module>
</module>
```

```
/**
 * @unknownTag value // OK
 */
@SpringBootApplication // no violations about unknown tag on class
public class Application {}

/**
 * @unknownTag value // OK
 */
@Configuration // no violations about unknown tag on class
class DatabaseConfiguration {}
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_six_checks.xml -o example_six_result.out ./src/ExampleSix.java
```

Result
```
开始检查……
检查完成。
```

**Comment by HCY123902** [comment]
> @HCY123902 you must run mvn clean verify locally to generate changes in those files, and you should always run this command and make sure it is passing before pushing to PR

yeah sure. I am using a Windows machine. Around 400 unit cases cannot pass locally, but these cases can pass on CI. I think there are no other issues reported from `mvn clean verify` aside from this

**Comment by romani** [comment]
Github, generate web site

**Comment by HCY123902** [comment]
Here are the screenshots of the examples on the generated documentation site

image
image
image

**Comment by HCY123902** [comment]
@romani Hello, can you kindly help to take a look at the pull request? Many thanks

**Comment by romani** [comment]
@HCY123902 , thanks a lot, but please try to use same java code for all examples to let user see as violation location is changing as you change config of Check. Yes, examples might be more verbose than now, but usually there are way to minimize java content.
trailing comment location only changing as marker of what is changing in behavior.

example: https://checkstyle.org/config_annotation.html#AnnotationLocation_Examples

**Comment by HCY123902** [comment]
> @HCY123902 , thanks a lot, but please try to use same java code for all examples to let user see as violation location is changing as you change config of Check. Yes, examples might be more verbose than now, but usually there are way to minimize java content. trailing comment location only changing as marker of what is changing in behavior.
> 
> example: https://checkstyle.org/config_annotation.html#AnnotationLocation_Examples

Thanks for the suggestion. I can change combine the first 5 examples into 1 only. The last example involving `allowedAnnotations` seems more appreciate to remain standalone though, since there is also a similar standalone example in [MissingJavadocType](https://checkstyle.org/config_javadoc.html#MissingJavadocType). Can I ask for your opinion on this

**Comment by romani** [comment]
ok, show that default value `Generated` is working fine.
and yes may be  there will be one more separate example for Spring classes.

**Comment by HCY123902** [comment]
> ok, show that default value `Generated` for allowedAnnotations is working fine. and yes may be there will be one more separate example for Spring classes.

Actually, I am thinking whether the description of `allowedAnnotations` is accurate here. It suppresses tag violations, but not missing documentation in my opinion

image

**Comment by HCY123902** [comment]
Here are the new examples. On line 17 of the common example, there is a tailing whitespace after the author tag. I will add a single line comment in Javadoc to indicate the existence of the whitespace. Please let me know what you think of this

Example results

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType"/>
    </module>
</module>
```

```
/**
 * @author a
 * @version 1.0
 */
public class ClassA { // OK
    /** */
    private class ClassB {} // OK
}

/**
 * @author 
 * @version abc
 */
public class ClassC {} // OK

/** */
public class ClassD<T> {} // violation, as param tag for <T> is missing

/** */
private class ClassE<T> {} // violation, as param tag for <T> is missing

/** */
@Generated
public class ClassF<T> {} // OK
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_one_checks.xml -o example_one_result.out ./src/ExampleOne.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleOne.java:17:1: 缺少 @param <T> 标签。 [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleOne.java:20:1: 缺少 @param <T> 标签。 [JavadocType]
检查完成。
```
---
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="scope" value="public"/>
        </module>
    </module>
</module>
```

```
/**
 * @author a
 * @version 1.0
 */
public class ClassA { // OK
    /** */
    private class ClassB {} // OK
}

/**
 * @author 
 * @version abc
 */
public class ClassC {} // OK

/** */
public class ClassD<T> {} // violation, as param tag for <T> is missing

/** */
private class ClassE<T> {} // OK

/** */
@Generated
public class ClassF<T> {} // OK
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_two_checks.xml -o example_two_result.out ./src/ExampleTwo.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleTwo.java:17:1: 缺少 @param <T> 标签。 [JavadocType]
检查完成。
```
---
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="authorFormat" value="\S"/>
        </module>
    </module>
</module>
```

```
/**
 * @author a
 * @version 1.0
 */
public class ClassA { // OK
    /** */
    private class ClassB {} // OK, as author tag check is ignored for inner class
}

/**
 * @author 
 * @version abc
 */
public class ClassC {} // violation, as author format with only whitespace or new line is invalid

/** */
public class ClassD<T> {} // violations, as param tags for <T> and author are missing

/** */
private class ClassE<T> {} // violations, as param tags for <T> and author are missing

/** */
@Generated
public class ClassF<T> {} // OK
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_three_checks.xml -o example_three_result.out ./src/ExampleThree.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleThree.java:14:1: 标签 @author 必须匹配： '\S' 。 [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleThree.java:17:1: 缺少 @author 标签。 [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleThree.java:17:1: 缺少 @param <T> 标签。 [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleThree.java:20:1: 缺少 @author 标签。 [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleThree.java:20:1: 缺少 @param <T> 标签。 [JavadocType]
检查完成。
```
---
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="scope" value="private"/>
            <property name="excludeScope" value="package"/>
        </module>
    </module>
</module>
```

```
/**
 * @author a
 * @version 1.0
 */
public class ClassA { // OK
    /** */
    private class ClassB {} // OK
}

/**
 * @author 
 * @version abc
 */
public class ClassC {} // OK

/** */
public class ClassD<T> {} // OK

/** */
private class ClassE<T> {} // violation, as param tag for <T> is missing

/** */
@Generated
public class ClassF<T> {} // OK
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_five_checks.xml -o example_five_result.out ./src/ExampleFive.java
```

Result
```
开始检查……
[ERROR] D:\RepoSense\try\.\src\ExampleFive.java:20:1: 缺少 @param <T> 标签。 [JavadocType]
检查完成。
```
---
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="allowedAnnotations" value="SpringBootApplication,Configuration"/>
        </module>
    </module>
</module>
```

```
/** */
@SpringBootApplication // no tag violations on class
public class Application<T> {}

/** */
@Configuration // no tag violations on class
class DatabaseConfiguration<T> {}
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_six_checks.xml -o example_six_result.out ./src/ExampleSix.java
```

Result
```
开始检查……
检查完成。
```

**Comment by romani** [comment]
Please do CLI in English.
```
var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java
```

**Comment by romani** [comment]
>It suppresses tag violations, but not missing documentation in my opinion

Share CLI output to share your concern, it might be defect

**Comment by HCY123902** [comment]
> Please do CLI in English.
> 
> ```
> var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
> /var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java
> ```
> 
> As we describe at https://github.com/checkstyle/checkstyle/blob/master/.github/ISSUE_TEMPLATE/bug_report.md

Thanks for the suggestion. I have changed the language for the example in the comment above

**Comment by HCY123902** [comment]
> > It suppresses tag violations, but not missing documentation in my opinion
> 
> Share CLI output to share your concern, it might be defect

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType"/>
    </module>
</module>
```

```
/** */
@Generated // violation suppressed for this class
public class Application<T> {}

/** */
public class Application<T> {} // violation, as param tag for <T> is missing

public class Application<T> {} // OK, as it does not have Javadoc
```

Command
```
java -jar ../checkstyle-10.7.0-all.jar -c example_one_checks.xml ./src/ExampleSeven.java
```

Result
```
Starting audit...
[ERROR] D:\RepoSense\try\.\src\ExampleSeven.java:6:1: Type Javadoc comment is missing @param <T> tag. [JavadocType]
Audit done.
Checkstyle ends with 1 errors.
```

It is not really a defect in my opinion. I think classes without Javadoc will not have violations in any case. The `allowedAnnotations` tag specifies which class with Javadoc can be ignored for the tag check

**Comment by romani** [comment]
Please keep pushing code for javadoc to let us discuss all at exact context of specific line.

**Comment by romani** [comment]
>It is more like the description may cause confusion of the reader.

Please update description as you think make sense and more clear and it as PR, we will review it, and most likely accept as opinion of users matter a lot.

It is not a defect. But I agree to expand description to be more explicit on behavior

**Comment by HCY123902** [comment]
> > It is more like the description may cause confusion of the reader.
> 
> Please update description as you think make sense and more clear and it as PR, we will review it, and most likely accept as opinion of users matter a lot.
> 
> It is not a defect. But I agree to expand description to be more explicit on behavior

yeah sure. Should I submit a separate issue and pull request for this

**Comment by romani** [comment]
>It is more like the description may cause confusion of the reader.

Please update description as you think make sense and more clear and it as PR, we will review it, and most likely accept as opinion of users matter a lot.

It is not a defect. But I agree to expand description to be more explicit on behavior

**Comment by romani** [comment]
Keep all in this PR.

**Comment by romani** [comment]
Github, generate web site

**Comment by romani** [comment]
Please generate web site as you finish changes.

Btw, you can do all that CI does, on your local https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-website-only

**Comment by HCY123902** [comment]
> Please generate web site as you finish changes.
> 
> Btw, you can do all that CI does, on your local https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-website-only

yeah sure. I have completed the modification. Here is how it looks on the site

image
image
image
image

**Comment by HCY123902** [comment]
Github, generate web site

**Comment by romani** [comment]
Please add config+java to show usage of all properties. 
Please show how each property causes violation and not .
Please remove from javadoc trailing comments to tags `// To reproduce`, you have comments for class - it is enough.

**Comment by HCY123902** [comment]
> Please add config+java to show usage of all properties. 

Do you mean adding 3 more use cases for `allowMissingParamTags`, `allowUnknownTags`, and `tokens` respectively?

> Please show how each property causes violation and not . 

Could you please elaborate more on this

> Please remove from javadoc trailing comments to tags `// To reproduce`, you have comments for class - it is enough.

yeah sure. I will change it

**Comment by romani** [comment]
>respectively?

Yes

>elaborate more on this

Show violation on version tag.

**Comment by HCY123902** [comment]
I have made the required adjustment. The version tag violations for configuration with `versionFormat` are now shown more clearly. I have to make subtle changes to the example for certain configurations to show the behavior difference since otherwise, the example will become too verbose and tedious.

Additionally. I notice that an unknown tag violation is reported on its own line and I have to add a separate comment for it in the `allowUnknownTags` example

**Comment by HCY123902** [comment]
Github, generate web site

**Comment by HCY123902** [comment]
@romani Hello, can you kindly take a look again? Many thanks

File: src/xdocs/config_javadoc.xml [hunk 1]
```diff
@@ -2181,24 +2293,126 @@ public class Test {
...
+ */
+public class ClassA { // OK
+  /** */
+  private class ClassB {} // OK
+}
 
-@Configuration // no violations about missing comment on class
-class DatabaseConfiguration {}
+/**
+ * @unknownTag value // OK, as unknown tag is allowed
```
Comment (romani) [inline src/xdocs/config_javadoc.xml:2380]: please put this tag in all (except for last) other examples, to let users see violation on it by default.

**Review by romani — CHANGES_REQUESTED** [review]
very good, last items:

File: src/xdocs/config_javadoc.xml (hunk 1 above)
Comment (HCY123902) [reply src/xdocs/config_javadoc.xml]: yeah sure. I have added it

Comment (HCY123902) [reply src/xdocs/config_javadoc.xml]: Github, generate web site

**Comment by romani** [comment]
Github, generate web site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java [hunk 2]
```diff
@@ -94,7 +97,7 @@
  * </li>
  * <li>
  * Property {@code allowedAnnotations} - Specify annotations that allow
- * missed documentation. Only short names are allowed, e.g. {@code Generated}.
+ * tag violations. Only short names are allowed, e.g. {@code Generated}.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java:100]: that allows skip validation at all.

**Review by romani — CHANGES_REQUESTED** [review]
Last:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java (hunk 2 above)
Comment (HCY123902) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java]: yeah sure. I ran it locally and `allowedAnnotations` does skip validation for every possible violation in this check

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocType">
            <property name="versionFormat" value="\$Revision.*\$"/>
            <property name="authorFormat" value="\S"/>
        </module>
    </module>
</module>
```

```
/**
 * @version invalid_version
 * @unknownTag value
 * @param K
 */
@Generated
public class ClassA<T> {
}
```

Command
```
java  -jar ../checkstyle-10.7.0-all.jar -c example_ten_checks.xml ./src/ExampleTen.java
```

Result
```
Starting audit...
Audit done.
```

Result without the @Generated annotation
```
Starting audit...
[ERROR] D:\RepoSense\try\.\src\ExampleTen.java:3:4: Unknown tag 'unknownTag'. [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleTen.java:4:4: Unused @param tag for 'K'. [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleTen.java:6:1: Type Javadoc comment is missing @author tag. [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleTen.java:6:1: Type Javadoc comment is missing @param <T> tag. [JavadocType]
[ERROR] D:\RepoSense\try\.\src\ExampleTen.java:6:1: Type Javadoc tag @version must match pattern '\$Revision.*\$'. [JavadocType]
Audit done.
Checkstyle ends with 5 errors.
```

**Comment by HCY123902** [comment]
Github, generate web site

**Review by romani — APPROVED** [review]
Thanks a lot for update!!

File: config/archunit-store/slices_should_be_free_of_cycles_suppressions [hunk 3]
```diff
@@ -799,7 +799,6 @@ Cycle detected: Slice api -> \
     - Method <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocMissingLeadingAsteriskCheck.isLastLine(com.puppycrawl.tools.checkstyle.api.DetailNode)> calls method <com.puppycrawl.tools.checkstyle.api.DetailNode.getText()> in (JavadocMissingLeadingAsteriskCheck.java:202)\
     - Method <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParagraphCheck.visitJavadocToken(com.puppycrawl.tools.checkstyle.api.DetailNode)> calls method <com.puppycrawl.tools.checkstyle.api.DetailNode.getType()> in (JavadocParagraphCheck.java:202)\
     - Method <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocParagraphCheck.visitJavadocToken(com.puppycrawl.tools.checkstyle.api.DetailNode)> calls method <com.puppycrawl.tools.checkstyle.api.DetailNode.getType()> in (JavadocParagraphCheck.java:203)\
-    - Constructor <com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTypeCheck.<init>()> calls constructor <com.puppycrawl.tools.checkstyle.api.AbstractCheck.<init>()> in (JavadocTypeCheck.java:204)\
```
Comment (rnveach) [inline config/archunit-store/slices_should_be_free_of_cycles_suppressions:802]: @romani Why did this change on a simple document change?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java [hunk 4]
```diff
@@ -94,7 +97,7 @@
  * </li>
  * <li>
  * Property {@code allowedAnnotations} - Specify annotations that allow
- * missed documentation. Only short names are allowed, e.g. {@code Generated}.
+ * skipping validation at all. Only short names are allowed, e.g. {@code Generated}.
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java:100]: What was wrong with original wording?

File: config/archunit-store/slices_should_be_free_of_cycles_suppressions (hunk 3 above)
Comment (romani) [reply config/archunit-store/slices_should_be_free_of_cycles_suppressions:802]: I am not sure, @HCY123902 , please confirm that this update is required by tool

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java (hunk 4 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java:100]: "Documentation" word is very confusing.

Comment (HCY123902) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java:100]: Here is the original discussion

https://github.com/checkstyle/checkstyle/pull/12736#issuecomment-1427245513

File: config/archunit-store/slices_should_be_free_of_cycles_suppressions (hunk 3 above)
Comment (HCY123902) [reply config/archunit-store/slices_should_be_free_of_cycles_suppressions:802]: I think this is caused by the additional lines in Javadoc, which move the line positions of the constructors and methods in  `JavadocTypeCheck.java`. This change is automatically done by `mvn clean verify`. Without it, CI will complain

**Comment by HCY123902** [comment]
@rnveach Hello, can you kindly take a look again? Many thanks

**Comment by romani** [comment]
Github, generate web site

**Comment by romani** [comment]
@HCY123902 , please help us to fix more issues, do not be blocked by this PR review.

**Comment by HCY123902** [comment]
> @HCY123902 , please help us to fix more issues, do not be blocked by this PR review.

yeah sure. However, I will appreciate it if this can be merged within 3 weeks as I will be quite busy after that

**Review by rnveach — APPROVED** [review]

**Pull request merged**